samples/glean/src/main/java/org/mozilla/samples/glean/GleanApplication.kt

```diff
@@ -67,7 +67,8 @@ class GleanApplication : Application() {
         nimbus = Nimbus(this,
             NimbusServerSettings(url)
         ).also { nimbus ->
-            nimbus.updateExperiments()
+            nimbus.fetchExperiments()
+            nimbus.applyPendingExperiments()
```

Comment:
I'm guessing that this doesn't await the fetch, not too big of a deal for the example app purposes. Considering this, what happens when apply is called while a fetch is active?  Would a returning fetch request in the middle of an apply have any effect?

Reply:
To follow up on this question, which I sort of answered myself is yes, whichever fetch or apply got to the opening of the underlying rkv writer first, would block the other thread until it was done.

Reply:
> Considering this, what happens when apply is called while a fetch is active? 
> Would a returning fetch request in the middle of an apply have any effect?

Both can be called from the main thread, without affecting performance. Yay!

If you called `fetch` then `apply`, then fetch goes and waits for the network, and writes to the experiment cache.
Before fetch can return, then the `apply` reads and clears the experiment cache. The `apply` can happen almost immediately after the client call to apply.

If we used the one thread, then you could call `fetch` then `apply` then the `apply` is run at after the `fetch` comes back, i.e. long after, and non-deterministically after you call the `apply`.

Reply:
This is doing main-thread IO, right? Is it worth calling that out?

Reply:
No. This is calling Rust `NimbusClient` from a `fetchScope` thread and a `dbScope` thread, so the app can call the kotlin `Nimbus` `fetch and `apply` safely from the main thread.